relevanceai/operations/dr/base.py

```diff
@@ -82,20 +91,108 @@ def fit_transform_documents(
             A list of documents with the original vector field and the new vector field.
 
         """
-        documents = list(self.filter_docs_for_fields([vector_field], documents))
+
+        documents = list(self.filter_docs_for_fields(vector_fields, documents))
         vectors = np.array(
-            self.get_field_across_documents(
-                vector_field, documents, missing_treatment="skip"
+            self.get_fields_across_documents(
+                vector_fields, documents, missing_treatment="skip"
             )
         )
+        vectors = vectors.reshape(-1, vectors.shape[1] * vectors.shape[2])  # hacky fix
         dr_vectors = self.fit_transform(vectors, dims=dims)
-        dr_vector_field_name = self.get_dr_vector_field_name(vector_field, alias)
-        self.set_field_across_documents(dr_vector_field_name, dr_vectors, documents)
+        del vectors  # free more memory, mainly for memory edgecases
+
+        dr_vector_field_name = self.get_dr_vector_field_name(vector_name, alias)
         if exclude_original_vectors:
-            dr_docs = self.subset_documents(["_id", dr_vector_field_name], documents)
-        return dr_docs
+            dr_docs = [{"_id": d["_id"]} for d in documents]
+            self.set_field_across_documents(dr_vector_field_name, dr_vectors, dr_docs)
+            return dr_docs
+        else:
+            self.set_field_across_documents(dr_vector_field_name, dr_vectors, documents)
+        return documents
+
+
+class PCA(DimReductionBase):
```

Comment:
We moved this to `relevanceai/operations/dr/models/`.

Reply:
could you remove what was moved @boba-and-beer